packages/client/src/forms/utils.ts

```diff
@@ -443,6 +445,22 @@ export const getFieldOptions = (
   }
 }
 
+/** Due to the large location trees with dependencies, generating options for them can be slow. We fix this by memoizing the options */
+export const getFieldOptions = memoize(
+  getFieldOptionsSlow,
+  (sectionName, field, values) => {
+    if (
+      field.type === SELECT_WITH_OPTIONS ||
+      field.type === DOCUMENT_UPLOADER_WITH_OPTION
```

Comment:
Does document uploader need this optimisation?

Reply:
The only one that really needs it is `SELECT_WITH_DYNAMIC_OPTIONS`.
So you could potentially only memoize in that case.
Would we want to refactor just for that?